nncf/torch/prepare_for_inference.py

```diff
@@ -0,0 +1,213 @@
+"""
+ Copyright (c) 2019-2023 Intel Corporation
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+      http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+import copy
+
+import numpy as np
+import torch
+from torch.quantization.fake_quantize import FakeQuantize
+
+from nncf.api.compression import CompressionAlgorithmController
+from nncf.torch.composite_compression import CompositeCompressionAlgorithmController
+from nncf.torch.nncf_network import NNCFNetwork
+from nncf.torch.pruning.filter_pruning.layers import FilterPruningMask
+from nncf.torch.pruning.operations import PT_PRUNING_OPERATOR_METATYPES
+from nncf.torch.pruning.operations import ModelPruner
+from nncf.torch.pruning.operations import PrunType
+from nncf.torch.quantization.layers import AsymmetricQuantizer
+from nncf.torch.quantization.layers import BaseQuantizer
+from nncf.torch.quantization.layers import SymmetricQuantizer
+
+SUPPORTED_ALGORITHMS = ["quantization", "filter_pruning"]
+
+
+def prepare_for_inference(
```

Comment:
I would expect to see it as a method of Controller. Potentially, it can replace its .export() API. For other backends, it could be a stub.